### ISSUE TITLE
Add new method to fetch the bowl ids from the logged in user's feed.

### ISSUE THREAD

**Issue by mukulbindal** [mukulbindal/fishbowlpy#15]
Fishbowl does not provide any API to fetch the bowl ids user is subscribed to. 

We need to use web scraping to find out the list of bowls a user is subscribed to, that usually comes into the feed.

We can easily use beautifulsoup for this. The challenge here is, the bowl Id is not present in that html page and somehow it is fetched from a javascript code which is probably obfuscated as per my observations.

Here we need to analyse how exactly normal webpage is fetching the bowls. After this is done we can think about web scraping which is easier part.

**Comment by eleensmathew** [comment]
Can i work on this?

**Comment by mukulbindal** [comment]
Yes please go ahead

**Comment by eleensmathew** [comment]
Is the code for fishbowl app api available anywhere? If fishbowl app api is a separate entity, how is the bowl id useful for us? Can you let me know what the bowl id should look like (I dont think it will be something like uuid so in what way do we want it?)

**Comment by mukulbindal** [comment]
Bowl Ids are usually same as bowl names but its not always true. For some bowls I had observed that its a bit different than bowl name which is why I need a concrete way to fetch all bowl Ids.

Bowl Id is required to fetch posts, so that is where its needed. You can try opening the bowls and observe the network tab of your browser to see how it really looks like. 

Last time I worked on it, I observed that its fetched via some jquery code which is not publicly visible, so that is a bit tricky part which needs to be solved.

**Comment by eleensmathew** [comment]
In this case 'tech india' can be the bowl id right? Also when I go to the link associated with that bowl eg https://www.fishbowlapp.com/bowl/tech-india we will use this link/url id to get the posts?

If this isn't what you expected, can you add an example of your thought process below?

<img width="797" height="334" alt="Image" src="https://github.com/user-attachments/assets/7e266b1e-d8d7-46f6-a32e-b12ecb66eae2" />

**Comment by mukulbindal** [comment]
Yes I meant 'tech-india' will be a bowl id, which is used in the url of that bowl. 

Generally for any bowl, if you make the bowl name all lower case and replace the spaces with hyphens (-), you get the bowl id. But there are few exceptions for which I need some concrete way, like how fishbowl web page fetches the list of bowl ids you are subscribed to.

**Comment by eleensmathew** [comment]
So, can you confirm that if I can get the link of the bowl names(https://www.fishbowlapp.com/bowl/tech-india) and extract the bowl id from that, it works? Am I missing something?
My thought process - Since you will probably scrape the posts from the link, then this bowl link will be useful right?

**Comment by mukulbindal** [comment]
Yes that works. But how are you planning to fetch the links?

**Comment by eleensmathew** [comment]
I see that there is a div called bowl link. Can't I get the link by scraping all these bowl-link tags?

<img width="645" height="189" alt="Image" src="https://github.com/user-attachments/assets/c532acb1-1cf9-4fd2-b99c-edc9dd5144f2" />

**Comment by mukulbindal** [comment]
I don't remember much. Please give it a try.

**Comment by eleensmathew** [comment]
So to reiterate, when the user logs in and switches to the feed section, you want to return an array of bowl IDs? Can the input to the function be a link to the feed, or does scraping happen when the user navigates to the feed?

**Comment by mukulbindal** [comment]
There shouldn't be any input ideally. You have the cookie, use it to open the feed and then do the scraping.

But you are free to add your approach, we can enhance it later if needed.

**Comment by eleensmathew** [comment]
I tried testing my code changes to include beautiful soup, but cloudfare is blocking my connection, which makes me unable to scrape the html from the file. Do you have any workarounds for this?

**Comment by mukulbindal** [comment]
Shouldn't happen ideally. Let me check today and get back if this library needs any update. I created it last year and unable to maintain the dependencies. 

By the way did you try to test existing code which is fetching posts? If that works, your changes should also work.

**Comment by mukulbindal** [comment]
@eleensmathew did you try any other way? I didn't get time to look into it

**Comment by eleensmathew** [comment]
hey I'll probably get a better look at the code tomorrow and then try debugging it. Thanks for helping out btw

**Comment by eleensmathew** [comment]
Hey i see that the get_posts() function is also not working. Pls let me know how to proceed
